Release 4 4 (#287)
mongodb/docs-node: source/whats-new.txt

Learn what's new in:

* :ref:`Version 4.4 <version-4.4>`
* :ref:`Version 4.3 <version-4.3>`
* :ref:`Version 4.2 <version-4.2>`
* :ref:`Version 4.1 <version-4.1>`
* :ref:`Version 4.0 <version-4.0>`
* :ref:`Version 3.6 <version-3.6>`

.. _version-4.4:

What's New in 4.4
-----------------

New features of the 4.4 Node.js driver release include:

> - KMIP support when using CSFLE.

Reviewer: Issue: KMIP to "KMIP provider". [Context](https://jira.mongodb.org/browse/DOCSP-20564)

```suggestion
- KMIP provider support when using CSFLE.
```

Author: I believe provider is implicit here...

- TLS support when using CSFLE.
- Hostname canonicalization now accepts "none", "forward", and "forwardAndReverse" as ``authMechanismProperties`` when using GSSAPI.

> - In the 4.0.0 release of the driver, the deprecated ``collection.count()`` method was inadvertently changed to behave like ``collection.countDocuments()``.

Reviewer: Comment: Can the 4.0 release notes mention this (possibly as an important admonition or warning)? Users of the 4.0-4.3 drivers will be unaware of this bug, however their drivers will be effected.

>   In this release, the ``collection.count()`` method is updated to match legacy behavior:

Reviewer: Issue: Passive voice ("is updated")

  - If a query is passed in, ``collection.count()`` will behave the same as ``collection.countDocuments()`` and perform a collection scan.

>   - If no query is passed in, ``collection.count()`` will behave the same as ``collection.estimatedDocumentCount()`` and rely on collection metadata..

Reviewer: Issues: Passive voice ("is passed"), and future tense ("will behave").

  .. important:: Deprecation Notice

>      The ``cursor.count()`` method is deprecated and will be removed in the next major version, along with ``collection.count()``.

Reviewer: Comment: Unsure which "next major version" is being referred to. Is this the Node.js driver, MongoDB Server, or both? My guess is driver (and possibly server), but this may be unclear to readers.

>      Use ``collection.estimatedDocumentCount()`` or ``collection.countDocuments()`` instead.

Reviewer: Issue: Consistently introduce methods. I think this applies elsewhere in this PR as well.

```suggestion
     Use the ``collection.estimatedDocumentCount()`` or ``collection.countDocuments()`` methods instead.
```

Learn what's new in:

* :ref:`Version 4.4 <version-4.4>`
* :ref:`Version 4.3 <version-4.3>`
* :ref:`Version 4.2 <version-4.2>`
* :ref:`Version 4.1 <version-4.1>`
* :ref:`Version 4.0 <version-4.0>`
* :ref:`Version 3.6 <version-3.6>`

.. _version-4.4:

What's New in 4.4
-----------------

New features of the 4.4 Node.js driver release include:

- KMIP provider support when using CSFLE.
- TLS support when using CSFLE.
- Hostname canonicalization now accepts "none", "forward", and "forwardAndReverse" as ``authMechanismProperties`` when using GSSAPI.
- In the 4.0.0 release of the driver, the deprecated ``collection.count()`` method was inadvertently changed to behave like ``collection.countDocuments()``.
  In this release, the ``collection.count()`` method is updated to match legacy behavior:

>   - If a query is provided, ``collection.count()`` behaves the same as ``collection.countDocuments()`` and perform a collection scan.
>   - If no query is not provided, ``collection.count()`` behaves the same as ``collection.estimatedDocumentCount()`` and relies on
>     collection metadata.

Reviewer: ```suggestion
  - If a query is provided, ``collection.count()`` behaves the same as ``collection.countDocuments()`` and performs a collection scan.
  - If no query is provided, ``collection.count()`` behaves the same as ``collection.estimatedDocumentCount()`` and relies on
    collection metadata.
```
